### PR TITLE
Expose response from Twitter API to client in provisioning API

### PR DESCRIPTION
**Pull request by smweber**

### CODE REVIEW

File: mautrix_twitter/web/provisioning_api.py [hunk 1]
```diff
@@ -108,8 +109,11 @@ async def login(self, request: web.Request) -> web.Response:
 
         try:
             await user.locked_connect(auth_token=auth_token, csrf_token=csrf_token)
+        except TwitterError as e:
+            self.log.debug("Failed to log in with TwitterError", exc_info=True)
+            raise web.HTTPUnauthorized(body=f'{{"error": "{e.message}"}}', headers=self._headers)
```
Comment (tulir) [inline mautrix_twitter/web/provisioning_api.py:114]:
```suggestion
            raise web.HTTPUnauthorized(body=json.dumps({"error": e.message}), headers=self._headers)
```

**Review by tulir — COMMENTED** [review]
What kinds of errors does it have here? 🤔

**Comment by smweber** [comment]
> What kinds of errors does it have here? 🤔

Specifically trying to dump this one:
```
File "/opt/mautrix-twitter/mautwitdm/errors.py", line 73, in check_error
    raise TwitterError(code, message)
mautwitdm.errors.TwitterError: 326: To protect our users from spam and other malicious activity, this account is temporarily locked. Please log in to https://twitter.com to unlock your account.
```

**Review by tulir — APPROVED** [review]
Not entirely sure if it's a good idea to expose all errors, but I guess we can hide errors later if needed

**Pull request merged**